### PR TITLE
Combine _.indexOf and _.lastIndexOf

### PR DESCRIPTION
**Pull request by jridgewell**
Exactly what the name implies: removes the duplication.

### CODE REVIEW

**Comment by megawac** [comment]
Mind forking some benchmarks and including them?

**Comment by jashkenas** [comment]
> removes the duplication

The most noble goal ;)

File: underscore.js [hunk 1]
```diff
@@ -603,40 +603,8 @@
...
-    if (item !== item) {
-      return _.findLastIndex(slice.call(array, 0, idx), _.isNaN);
-    }
-    while (--idx >= 0) if (array[idx] === item) return idx;
-    return -1;
-  };
-
   // Generator function to create the findIndex and findLastIndex functions
-  function createIndexFinder(dir) {
+  function createPredicateIndexFinder(dir) {
```
Comment (akre54) [inline underscore.js]: Why do we have NFEs here? This seems to be recent (`createReduce` too). Older style (`group` specifically) was to assign to a var.

Comment (michaelficarra) [inline underscore.js]: That's an FD, not an NFE.

Comment (akre54) [inline underscore.js]: whatever. pedantry aside, can we swap this back to the style of the rest of the code? It looks ugly.

**Pull request merged**